# Identitas Penulis
- **Nama Lengkap**: Muhammad Ikctiar Saputra
- **NIM**: 250401020169
- **Kelas**: IF401
- **Program Studi**: PJJ Informatika

In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler

URL = "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv"
df = pd.read_csv(URL)
print("Dataset shape:", df.shape)
print(df.head())

Dataset shape: (244, 7)
   total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4


In [2]:
# Langkah 2: Handling Missing Values
print("Null values count in each column:")
print(df.isnull().sum())

Null values count in each column:
total_bill    0
tip           0
sex           0
smoker        0
day           0
time          0
size          0
dtype: int64


In [3]:
# Langkah 3: Encoding Kategorikal
# Mengubah tipe data kategori ('sex', 'smoker', 'day', 'time') menjadi representasi biner
# drop_first=True digunakan untuk menghindari jebakan multikolinieritas (dummy variable trap)
df_encoded = pd.get_dummies(df, columns=['sex', 'smoker', 'day', 'time'], drop_first=True)
print("Shape setelah One-Hot Encoding:", df_encoded.shape)
print("\nKolom baru hasil encoding:")
print(df_encoded.columns.tolist())

Shape setelah One-Hot Encoding: (244, 9)

Kolom baru hasil encoding:
['total_bill', 'tip', 'size', 'sex_Male', 'smoker_Yes', 'day_Fri', 'day_Sat', 'day_Sun', 'time_Dinner']


In [4]:
# Langkah 4: Train-Test Split
# Memisahkan fitur dan target. Target kita adalah memprediksi 'tip' pelanggan
X = df_encoded.drop(columns=['tip'])
y = df_encoded['tip']

# Membagi data menjadi 80% untuk pelatihan dan 20% untuk pengujian
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

print("Ukuran X_train:", X_train.shape)
print("Ukuran X_test:", X_test.shape)
print("Ukuran y_train:", y_train.shape)
print("Ukuran y_test:", y_test.shape)

Ukuran X_train: (195, 8)
Ukuran X_test: (49, 8)
Ukuran y_train: (195,)
Ukuran y_test: (49,)


In [5]:
# Langkah 5: Feature Scaling
# Penskalahan hanya dilakukan pada variabel numerik kontinu ('total_bill') menggunakan StandardScaler
# Fit dilakukan hanya pada X_train, sedangkan X_test cukup di-transform agar tidak terjadi data leakage
scaler = StandardScaler()

X_train_scaled = X_train.copy()
X_test_scaled = X_test.copy()

# Penskalaan kolom 'total_bill'
X_train_scaled[['total_bill']] = scaler.fit_transform(X_train[['total_bill']])
X_test_scaled[['total_bill']] = scaler.transform(X_test[['total_bill']])

print("Rata-rata total_bill setelah scaling (Train):", np.round(X_train_scaled['total_bill'].mean(), 2))
print("Standar deviasi total_bill setelah scaling (Train):", np.round(X_train_scaled['total_bill'].std(), 2))

Rata-rata total_bill setelah scaling (Train): 0.0
Standar deviasi total_bill setelah scaling (Train): 1.0


## Kesimpulan
Pada pertemuan keenam ini, saya mempelajari langkah-langkah penting dalam prapemrosesan data sebelum masuk ke tahap pelatihan model machine learning:
1. Memeriksa missing values sebagai langkah wajib awal sebelum transformasi.
2. Melakukan One-Hot Encoding dengan fungsi `pd.get_dummies(drop_first=True)` untuk mengubah variabel bertipe kategori/string menjadi angka biner sehingga dapat dibaca oleh algoritma matematika machine learning.
3. Membagi data menjadi subset latih (train) dan subset uji (test) secara acak menggunakan `train_test_split()`. Pemisahan ini penting agar kita memiliki data independen untuk mengevaluasi kinerja model.
4. Melakukan penormalan skala fitur numerik kontinu (`total_bill`) dengan `StandardScaler` agar memiliki rata-rata 0 dan standar deviasi 1. Hal ini mencegah dominasi fitur berangka besar terhadap fitur berangka kecil.

**Temuan Utama**: One-hot encoding berhasil mengekspansi fitur dataset Tips menjadi 9 kolom. Pembagian train-test split berhasil membagi data dengan proporsi 195 baris untuk pelatihan dan 49 baris untuk pengujian.

**Keterbatasan/Pertanyaan**: Bagaimana cara menghubungkan dataset yang telah siap diproses ini untuk dimasukkan ke dalam algoritma Regresi Linier dan mengevaluasi hasil prediksinya?